syntax="proto3";

package System;

service EnergyCalculation {
  rpc GetTask(GetTaskRequest) returns (GetTaskResponse) {}
  rpc PutResult(PutResultRequest) returns (PutResultResponse) {}
}

enum ResponseStatus {
  SUCCESS = 0;
  TRY_LATER = 1;
  INTERNAL_ERROR = 2;
}

message GetTaskRequest {
  string client_identifier = 1;
}

// task_content is json stringified the task data and task_identifier matches with md5 hash of task_content
message GetTaskResponse {
  ResponseStatus status = 1;
  string task_identifier = 2;
  string task_content = 3;
}

message PutResultRequest {
  string task_identifier = 1;
  string results = 2;
}

message PutResultResponse {
  ResponseStatus status = 1;
}
